File: examples/persian_hex.c
#include <stdio.h>
#include "persian_hex.h"

int main() {
    int number;
    printf("شماره وارد کنید: ");
    scanf("%d", &number);
    if (number < 0) {
        printf("خطا: عدد باید غیر منفی باشد.\n");
    } else {
        show_persian_hex(number);
    }
    return 0;
}

File: examples/persian_hex.py
import os
import sys
from pathlib import Path

parent_dir = Path(__file__).resolve().parent.parent
sys.path.append(str(parent_dir))
sys.path.append(os.path.join(parent_dir, "libs"))

from persian_hex import PersianHex

def main():
    try:
        number = int(input("شماره وارد کنید: "))
        if number < 0:
            print("خطا: عدد باید غیر منفی باشد.")
        else:
            PersianHex(number).show()
    except ValueError:
        print("خطا: لطفاً یک عدد صحیح معتبر وارد کنید.")

if __name__ == "__main__":
    main()
